presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -462,116 +366,265 @@ else if (!result.getBlocked().isDone()) {
 
     private boolean isReadyForExecution(StreamingPlanSection section)
     {
-        SqlStageExecution stageExecution = getStageExecution(section.getPlan().getFragment().getId());
-        if (stageExecution.getState() != PLANNED) {
+        Optional<SectionExecution> sectionExecution = getLatestSectionExecution(getStageId(section.getPlan().getFragment().getId()));
+        if (sectionExecution.isPresent() && (sectionExecution.get().isRunning() || sectionExecution.get().isFinished())) {
```

Comment:
Sure.  Is this just for readability?

Reply:
@rschlussel  Correct~

Reply:
just realized we can't do that because we don't want to return false if section execution is absent.  if section execution is absent, it means it hasn't been scheduled yet, so we actually always need to do the full check for those about whether its children have finished.